### Bayesian Statistics with conditional probability


People rely on the collective intelligence of previous experiences to protect themselves or take better decisions in future like save themselves from eating bad food at the wrong restaurant or avoid stunning flops at the theater etc. 

We discussed conditional probability of an event as the probability obtained using additional information that some other event has already occurred. We used the following formula for finding P(B|A):

                                             P(A ∩ B)  
                                    P(B|A) = --------  
                                               P(A)

Where occurance of event B is dependent on event A.    

In this notebook we extend the discussion of conditional probability to applications of Bayes' theorem (or Bayes' rule). Bayes ruke is used to updates the prior probabilities based on additional information that is obtained later. Bayes theorem deals with sequence of events where each occurance of a subsequent event provides new information that is used to revise the probability of the previous event. The terms prior probability and posterior probability are the commonly appearing terms in Bayes theorem discussion.

**prior probability** is an initial probability value originally obtained before any additional information is obtained.

**posterior probability** is a probability value that has been revised by using additional information that is later obtained.

### Bayes' Theorem

The probability of event A, given that event B has subsequently occurred, is mathematically represented as below

                                             P(A) * P(B/A)
                        P(A/B) = -------------------------------------
                                  [P(A) * P(B/A)] + [ P(A') * P(B/A')]
                                  

Consider an example of conducting cancer test. Tests detect things that don’t exist (false positive), and miss things that do exist (false negative). People often consider the test results directly, without considering the errors in the tests. Bayes’ theorem converts the results from a test into the real probability of the event. 

**Correct for measurement errors.** If you know the real probabilities and the chance of a false positive and false negative, you can correct for measurement errors.

**Relate the actual probability to the measured test probability.** Bayes’ theorem lets you relate Pr(A|X), the chance that an event A happened given the indicator X, and Pr(X|A), the chance the indicator X happened given that event A occurred. Given mammogram test results and known error rates, you can predict the actual chance of having cancer.

Bayes’ Theorem. It lets you take the test results and correct for the “skew” introduced by false positives. Consider the example of cancer test again to illustrate what Bayes formula is doing.

Let 'A' be the event of person having cancer.
Let 'X' be the event of positive test

Pr(A|X) = Chance of having cancer (A) given a positive test (X). This is what we want to know: How likely is it to have cancer with a positive result?

Pr(X|A) = Chance of a positive test (X) given that you had cancer (A). This is the chance of a true positive.

Pr(A) = Chance of having cancer.

Pr(not A) = Chance of not having cancer.

Pr(X|not A) = Chance of a positive test (X) given that you didn’t have cancer (~A). This is a false positive.

It all comes down to the chance of a true positive result divided by the chance of any positive result. We can simplify the equation to:

                      P(X/A) * P(A)
            P(A/X) = ---------------
                          P(X)

Pr(X) is a normalizing constant and helps scale our equation. Pr(X) tells us the chance of getting any positive result, whether it’s a real positive in the cancer population or a false positive in the non-cancer population. It’s a bit like a weighted average, and helps us compare against the overall chance of a positive result.

Below example illustrates the formula. 

----
`1. Consider an example:` In Boone County, 51% of the adults are males. One adult is randomly selected for a survey involving credit card usage. What is the prior probability that the selected person is a male.

**Solution: ** Its given that 51% of the adults in the county are males. Consider 'm' as an event of selecting an adult. Then the probability of randomly selecting an adult and getting a male is given by P(m) = 0.51

`2. Consider another example:` It is later learned that the selected survey subject was smoking a cigar. Also, 9.5% of males smoke cigars, whereas 1.7% of females smoke cigars (based on data from the Substance Abuse and Mental Health Services Administration). Use this additional information to find the probability that the selected subject is a male.

**Solution: ** Based on the additional given information, we have the following:
    
    Let c denote the event that the adult smokes cigar
        
    c' is compliment event of c that represents adult not smoking a cigar
        
    P(m) = 0.51 because 51% of the adults are males
    
    P(m') = 0.49 because 49% of the adults are females (not males)
    
    P(c|m) = 0.095 because 9.5% of the males smoke cigars (That is, the probability of getting someone who smokes cigars, given              that the person is a male, is 0.095.)

    P(c|m') = 0.017 because 1.7% of the females smoke cigars (That is, the probability of getting someone who smokes cigars,                  given that the person is a female, is 0.017)

Applying Bayes' theorem which outlined above, we get the following result:

                                               p(m) * p(c/m)
                        P(m | c) = --------------------------------------
                                    [p(m) * p(c/m)] + [ p(m') * p(c/m')]
                                    
                                 =         0.51 * 0.095
                                   -------------------------------
                                   (0.51 * 0.095) + (0.49 * 0.017)
                                   
                                 =  0.853
                                

Before we knew that the survey subject smoked a cigar, there is a 0.51 probability that the survey subject is male. But after realizing that the subject smoked a cigar, the probability is revised to 0.853. There is a 0.853 probability that the cigar−smoking respondent is a male. The likelihood of a male increased dramatically with the additional information that the subject smokes cigars.

Lets apply Bayes theorem on a multivariate dataset to find insights. Load the framingham data from the directory '/datasets/framingham' ... This data is from the Framingham Heart Study : https://www.framinghamheartstudy.org

In [3]:
framingham_data <- read.csv("../../../datasets/framingham/framingham.csv")
head(framingham_data)

male age education currentSmoker cigsPerDay BPMeds prevalentStroke
1 1    39  4         0              0         0      0              
2 0    46  2         0              0         0      0              
3 1    48  1         1             20         0      0              
4 0    61  3         1             30         0      0              
5 0    46  3         1             23         0      0              
6 0    43  2         0              0         0      0              
  prevalentHyp diabetes totChol sysBP diaBP BMI   heartRate glucose TenYearCHD
1 0            0        195     106.0  70   26.97 80         77     0         
2 0            0        250     121.0  81   28.73 95         76     0         
3 0            0        245     127.5  80   25.34 75         70     0         
4 1            0        225     150.0  95   28.58 65        103     1         
5 0            0        285     130.0  84   23.10 85         85     0         
6 1            0        228     180.0 110   30.30 77         99     0

In [8]:
with(framingham_data,table(currentSmoker,TenYearCHD))

             TenYearCHD
currentSmoker    0    1
            0 1834  311
            1 1762  333

**Question: ** What is the probability for a person to have coronary heart disease condition who is a smoker?

According to the Bayes theorem formula, lets define events.

**Solution **: Let c be an event of selecting a current smoker and c' be event of selecting a non current smoker

d be the event of person having risk of coronory heart disease and d' be event of person not having risk of coronory heart disease

                    p(d) * p(c/d)
     p(d/c) =  -------------------------------------
               [p(d) * p(c/d)] + [ p(d') * p(c/d')]
            

Pr(d|c) = Chance of having coronary heart disease (d) given a person is a smoker (c). This is what we want to know: How likely is it to have heart disease if person smokes? 

Pr(c|d) = Chance of a person being a smoker (c) given that he has coronary heart disease (d). This is the chance of a true positive 0.517

Pr(d) = Chance of having coronary heart disease (311+333)/(1762+333+1834+311)  = 644/4240 = 0.15

Pr(d') = Chance of not having coronary heart disease (1834+1762)/(1762+333+1834+311) = 3596/4240 = 0.85

Pr(c|not d) = Chance of person being smoker (c) given that he doesn't have the disease (d'). This is a false positive 0.49

p(c)  = (1762+333)/(1762+333+1834+311)  = 2095/4240 = 0.4941


              P(c ∩ d)      333
    p(c/d) =  --------   = ----- = 0.517
                p(d)        644


                P(c ∩ d')       1762
    p(c/d') =  -----------   = ------ = 0.49
                 p(d')          3596

                         p(d) * p(c/d)               
    p(d/c) =   ---------------------------------------
                    [p(d) * p(c/d)] + [ p(d') * p(c/d')]


                     0.15 * 0.517            
           =   -----------------------------------
                  (0.15 * 0.517) + (0.85 * 0.49)
         
           =    0.1569
        

Use simplified version of Bayes rule and compare the result.

                                      p(d) * p(c/d)
                        P(d/c)  =  -------------------
                                         p(c)
                                         
                                     0.15 * 0.517
                                =  ---------------
                                         0.4941
                                         
                                =  0.1569